composer/models/huggingface.py

```diff
@@ -469,7 +469,6 @@ def eval_forward(self, batch, outputs: Optional[Any] = None):
             self.labels = batch.pop('labels')
             generation = self.generate(batch['input_ids'],
                                        attention_mask=batch['attention_mask'],
```

Comment:
this should be done in a backards compatible way. so, get generation length from the batch, if it exists, raise deprecation warning saying to use max_new_tokens instead, and `max_new_tokens = batch.get('generation_kwargs', {}).get('max_new_tokens', batch.get('generation_length', None))` or something like that

Reply:
This was never set explicitly in any of our icl datasets. This is for anyone who's written custom icl_datasets, correct?

Reply:
ya